txjuju/cli.py

```diff
@@ -13,9 +13,45 @@
 from twisted.internet.protocol import ProcessProtocol
 from twisted.python import log
 
+from . import config
 from .errors import CLIError
 
 
+class BootstrapSpec(object):
+    """A specification of the information with which to bootstrap."""
+
+    DEFAULT_SERIES = "trusty"
+
+    def __init__(self, name, type, default_series=None, admin_secret=None):
+        """
+        @param name: The name of the controller to bootstrap.
+        @param type: The provider type to use.
+        @param default_series: The OS series to provision by default.
+            If not provided, it defaults to trusty.
+        @param admin_secret: The password to use for the admin user,
+            if any.
+        """
+        if default_series is None:
+            default_series = self.DEFAULT_SERIES
+
+        self.name = name
+        self.type = type
```

Comment:
I don't like seeing python reserved words used as variable names, can we change this parameter to _type or provider_type? I don't mind so much when the class attribute is named type.